### ISSUE TITLE
Breaking LoS converts forcefire or attack orders into move orders (Bug)

### ISSUE THREAD

**Issue by FiveAces** [OpenRA/OpenRA#14734]
Initially, I wanted to title this as a bug with forcefire, but after some testing it turns out that this applies to all attack modes: If I assign one of my units a target that I have vision of and the line of sight breaks (spy plane timer runs out, airplane idling circles etc), the attack order gets converted into a move order that I never issued. This is obnoxious especially when playing a V2/Yak combo that relies on getting good hits onto an enemy infantry army: I move the Yak in to spot the army, assign my V2s a target via ctrl (forcefire) and move the Yak out of the danger zone again. The expected behaviour for my V2s would be to forcefire the spot I ordered them to, but what happens instead is that they start moving headfirst into their demise after I move the Yak away and thus break line of sight.

I would have an idea for a relatively simple fix: Can we make it so that the forcefire command always targets the cell that I click on instead of an actor? This would prevent two bugs from happening: The one mentioned above, and the one where you can't forcefire camo pillboxes after they have re-cloaked even if you know their exact position.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/26001964/34941980-37ee7134-f9f6-11e7-8db5-d60e48f2e302.gif)

**Comment by klaas1** [comment]
Force-fire being directed at a location is the most logical behavior anyway, so this would fix part of the issue.

This is somewhat related to the longstanding "Attack order becomes Move order on loss of vision", for example in #10223. 

I still think this hasn't been fixed properly, and the logical behavior would be to cancel direct attack orders on moving actors when vision is lost.

**Comment by pchote** [comment]
Is this a new regression, or just the usual old broken behavior?

If the latter, then this issue goes right to the deepest parts of OpenRA, and there are no simple fixes.  A lot of work went into the latest dev cycle towards fixing this (#13995, #14189, #14295, #14537, #14561, #14558, #14608, #14619), but a lot more still needs to happen before any benefits will be visible ingame.

**Comment by FiveAces** [comment]
@pchote It is the old bug all over again, I just wanted to file this PR since I thought my fix might be easily applicable (ctrl targeting a cell instead of an actor). If this is not the case, then we're in for the long haul.

**Comment by GraionDilach** [comment]
PR is a Pull Request asin a patch. What you filed is referred as a ticket or an issue.

**Comment by pchote** [comment]
Issue confirmed - the regression here is that this playtest is the first time we have had units with a minimum weapon range greater >= vision range.

This leads to bogus behaviours like: https://www.dropbox.com/s/l2v2g5wzqaxyrby/v2.mov?dl=0 (note that this gives us a simple repro case for #10224).  See also [this forum post](http://www.sleipnirstuff.com/forum/viewtopic.php?f=82&p=306303#306303).

We're going to need to do something about this - either increase the vision range or revert the minimum weapon range to bring back the two-cell overlap.  Adding to the milestone to track the mod-level fix to the specific v2/artillery behaviour regressions in the playtest.

**Comment by pchote** [comment]
Closing as a combination of fixed by #14753 (MinRanges) / wontfix (forcing force-attack against terrain) / duplicate of other issues.
